### DISCUSSION TITLE
Assignable midi out channel for each pad (work around for triggering external drum machines)

### DISCUSSION THREAD

**Discussion by 777paranoia** [bangcorrupt/hacktribe#179] [Ideas]
Hey, kinda struggling to find a concise way to articulate this, but one of the biggest issues I have using the e2 to sequence other older drum machines is with each pad's sequence being locked to a corresponding midi channel, and the fact that virtually all of my older drum machines have voices assigned to keyboard note scale values on a single midi channel. Sequencing from a single pad channel on the e2 in poly mode is obviously an option but not a good one, considering the velocity is confined to each step and not each note on a step, which ultimately renders useless for every application I've tried with it. 

What are the chances of being able to assign which midi channel the midi information from each pad's sequence outputs to? 

Like for example; 
-Say my external drum machine is expecting midi on channel 1, with its kick drum voice on C4 + snare on C#4 + tom on D4 + hihat on D#4.
-On Part 5 of the E2, I write a (mono) sequence consisting of C4 note values, each with their own velocity and gate time values. On Part 6, another sequence for C#4, and so on with Pt7=D4, Pt8=D#4.
-Then, for Parts 5-8 (as well as the rest of the parts); there would be an option to assign each of these individual step sequences to midi ch 1, and as a result, the external drum machine receives each specific note value + their velocity/gate values on the same midi channel. Same could be done on any part with ch's 1-16.

Hope that makes sense. Being able to do single-midi channel note scale sequencing from the E2 would bring it up to par with the Beatstep Pro in terms of what I need from them bc of what the E2 lacks, and personally speaking would totally consummate my live set up so hope there's some potential here. Thanks!

**Comment by bangcorrupt** [comment]
I can add MIDI filtering and remapping at some point.  It would also be useful to split input key ranges to different channels.  Maybe matrix controls too.

**Comment by playbackero** [comment]
@bangcorrupt first of all, let me say THANK YOU for doing this hack (I read it as creativity against corporate greed, because it helps people and exposes self crippling practices from Korg).

I'm with 777paranoia on this one. The Electribe has an amazing X0X style sequencer but it cannot be used with non-configurable machines (older ones) that expect to **receive different notes on the same midi channel.** It would be amazing if the **channel assignment per pad** could be changed, and also the **note that the pad sends when set to trigger.**

If you can and just for kicks, take a look at the AKAI S20, I think a lot of the voice layout and sampler workflow queues were taken from it, as it has 16 pads that when pressed have a fixed trigger note, but if you plug a keyboard, each pad can be played on all keyboard notes, the only thing is that the internal sequencer SUCKS! but the sound engine has a  peculiar sound and it is sweet :)
All of the matrix options apply for each pad/button (except DISK).

Thanks again!

![IMG_2384](https://user-images.githubusercontent.com/112127953/186771103-13193ed5-8867-4a5f-bc13-173ae59b1bd9.JPG)
